### ISSUE TITLE
Phase 2: Kornia GPU augmentation support for segmentation masks

### ISSUE THREAD

**Issue by farukalamai** [roboflow/rf-detr#997]
### Search before asking

- [x] I have searched the RF-DETR [issues](https://github.com/roboflow/rf-detr/issues) and found no similar feature requests.

### Description

# Kornia GPU augmentation support for segmentation masks

## Summary

Extend the GPU augmentation backend added in #874 to work with segmentation models. Currently, when `segmentation_head=True`, GPU augmentation is disabled and the pipeline silently falls back to CPU Albumentations — even if the user explicitly sets `augmentation_backend="gpu"` or `"auto"`.

## Motivation

Phase 1 (#862, #874) shipped GPU-accelerated augmentation for detection training via Kornia. Segmentation was explicitly deferred as "Phase 2" because instance mask augmentation wasn't wired through the Kornia pipeline.

For segmentation training runs, CPU-side Albumentations is often the dominant per-batch cost — especially when geometric transforms are applied to many per-instance masks. Enabling the GPU path for segmentation would close the last major gap from the Phase 1 design and give segmentation users the same throughput benefits detection users already have.

## Current behavior

```python
model = RFDETRSegPreview()
model.train(
    dataset_dir="path/to/dataset",
    augmentation_backend="gpu",   # silently ignored
)
```

Two places force the fallback:

- `src/rfdetr/datasets/coco.py` — when `include_masks=True`, forces `augmentation_backend="cpu"` and emits a warning.
- `src/rfdetr/training/module_data.py` — `on_after_batch_transfer` returns the batch unchanged with a one-time warning: *"Kornia GPU augmentation skipped for segmentation models (phase 2)"*.

## Expected behavior

`augmentation_backend="gpu"` and `"auto"` should work the same way for segmentation models as they do for detection models. Images, boxes, and per-instance masks should be augmented in sync on the GPU, with instances dropped consistently when a box/mask is removed by augmentation (e.g. rotated out of frame).

## Scope

- Mask support in the Kornia augmentation pipeline (image + boxes + masks stay in sync).
- No new augmentation types — reuses the existing Phase 1 transform registry.
- No changes to the mask loss; it's resolution-agnostic and unaffected.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!
